core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -0,0 +1,233 @@
+import tc.oc.util.bukkit.component.types.PersonalizedTranslatable;
+import tc.oc.util.bukkit.nms.NMSHacks;
+
+@ListenerScope(MatchScope.RUNNING)
+public class StatsMatchModule implements MatchModule, Listener {
+
+  public static class Factory implements MatchModuleFactory<StatsMatchModule> {
+
+    @Override
+    public StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+      return new StatsMatchModule();
+    }
+  }
+
+  public static class playerStats {
+    private int kills = 0;
+    private int deaths = 0;
+    private int killstreak = 0;
+    private int killstreakMax = 0;
+    private int longestBowKill = 0;
+
+    public void onMurder() {
+      kills++;
+      killstreak++;
+      if (killstreak > killstreakMax) killstreakMax = killstreak;
+    }
+
+    public void onDeath() {
+      deaths++;
+      killstreak = 0;
+    }
+
+    public void setLongestBowKill(double distance) {
+      if (distance > longestBowKill) {
+        longestBowKill = Integer.parseInt(new DecimalFormat("#").format(Math.round(distance)));
+      }
+    }
+
+    public Component getBasicStatsMessage() {
+      String KD;
+      if (deaths == 0) {
+        KD = "0";
+      } else {
+        DecimalFormat df = new DecimalFormat("#.##");
+        KD = df.format(kills / deaths);
+      }
+      return new Component(
+          new PersonalizedTranslatable(
+                  "stats.basic",
+                  new PersonalizedText(Integer.toString(kills), ChatColor.GREEN),
+                  new PersonalizedText(Integer.toString(killstreak), ChatColor.GREEN),
+                  new PersonalizedText(Integer.toString(deaths), ChatColor.RED),
+                  new PersonalizedText(KD, ChatColor.GREEN))
+              .render());
+    }
+  }
+
+  public static Map<String, playerStats> allPlayerStats;
+
+  public StatsMatchModule() {
+    allPlayerStats = new HashMap<>();
+  }
+
+  @EventHandler
+  public void onPlayerDeath(MatchPlayerDeathEvent event) {
+    MatchPlayer victim = event.getVictim();
+    MatchPlayer murderer = null;
+    try {
+      murderer = event.getKiller().getParty().getPlayer(event.getKiller().getId());
+    } catch (NullPointerException ignored) {
+    }
+
+    playerStats victimStats = allPlayerStats.get(event.getVictim().getBukkit().getName());
+    if (victimStats == null) {
+      allPlayerStats.put(victim.getBukkit().getName(), new playerStats());
+      victimStats = allPlayerStats.get(event.getVictim().getBukkit().getName());
+    }
+    victimStats.onDeath();
+
+    playerStats finalVictimStats = victimStats;
+    // Without this delay the "Press LShift to dismount" message from the death minecart will
+    // override the stats message.
+    event
+        .getMatch()
+        .getScheduler(MatchScope.LOADED)
+        .runTaskLater(
+            0,
+            () -> {
+              NMSHacks.sendHotbarMessage(
+                  victim.getBukkit(), finalVictimStats.getBasicStatsMessage());
+            });
```

Comment:
Make this a repeating task, send the message for a second or so or else it fades away too quick

Reply:
How do i make it repeat? I tried `#runTaskTimer` with a delay of 0 and interval of 5, but that looked like a infinite thing, right?

Reply:
i'm not sure with the scheduler. I know bukkit has it's own runRepeatingTask but idk why the match scheduler would not expose it

Reply:
Yeah that is exposed with the scheduler through `runTaskTimer` But that is neverending! How do i stop it o_o

Reply:
It returns an int for the task id. you can cancel task by it's id

Reply:
Yeah, but i have to cancel the task through `Bukkit#getScheduler#cancelTask`
`Match#getScheduler(MatchScope.LOADED)#cancelTask` does not exist.

Reply:
And the normal `runTaskTimer` does not return an int, you have to do `#getTaskId` it seems